doc/creating_planner_plugin/emptyplan_planner_tutorial.rst

```diff
@@ -0,0 +1,42 @@
+Motion Planning Plugin Template
+===============================
+In this section, we will show how to add a new motion planner to MoveIt as a plugin. This is going to be a template planner which has the basics to run as a plugin for MoveIt. It does not actually do any sort of planning but it is just a start point for adding new planners.
+
+`This <http://wiki.ros.org/pluginlib>`_ gives a detailed explanantion of how to add plugins in ROS in general. The two necessary elements are base and plugin classes. The plugin class inherits from the base class and overrides its virtual functions. In MoveIt, ``planning_interface`` is defined as the base class from which any new planner plugin should inherit. For demonstration purposes, let's call the new planner ``emptyplan``. Create a folder with this name in ``moveit/moveit_planners`` directory; this is where all the files related to ``emptyplan`` plugin should live in. The following graph shows a brief overall view of the relation between packages and classes for adding a new template planner in MoveIt.
```

Comment:
I wouldn't have the reader create their planner within the moveit repo since it will affect the git. It would be just as easy for them to create their own package within the src directory of their catkin ws